pkg/auth/filter.go

```diff
@@ -153,6 +156,7 @@ func ToMiddleware(auth Authenticator) Middleware {
 						"system:cattle:error",
 					},
 				}
+				ctx = request.WithValue(ctx, CattleAuthFailed, "true")
 			} else if !ok {
 				info = &user.DefaultInfo{
```

Comment:
So all this PR does is add a single key/val to the context if the authentication failed. But then below if authentication worked but user is unauthenticated we don't care, because they are using a rancher token ?